students/Daniel_Spray/Lesson04/mailroom2.py

```diff
@@ -1,135 +1,119 @@
-def main(b=(["William Gates, III","Mark Zuckerberg","Jeff Bezos","Paul Allen"],[653784.49,16396.10,877.33,708.42], [2,3,1,3], [], [])):
-    print("""This program will hopefully help you send some meaningful messages")
+import sys
 
+donation_data = {
+'William Gates, III': [100000.00,553784.49],
+'Mark Zuckerberg': [5000.00,5000.00,6396.10],
+'Jeff Bezos': [877.33],
+'Paul Allen': [100.00,100.00,508.42]
+}
+
+def menu():
+    selection = input("""This program will hopefully help you send some meaningful messages
 Type the corresponding number to select from the following list:
 
 1: Send a Thank You
 2: Create a Report
-3: Send to All
+3: Send Thank You to Everyone
 4: Quit
-""")
-    g=str(input(">"))
-	
+>""")
+
     switch_menu = {
-        '1': saty,
-        '2': car,
+        '1': send_thank_you,
+        '2': create_report,
         '3': send_all,
-        '4': quit,
-        '5': try_again
+        '4': quit
     }
+    
+    while selection != '1' and selection != '2' and selection != '3' and selection != '4':
```

Comment:
you can do `selection not in ('1', '2', '3', '4')` but I would recommend doing `selection not in switch_menu`